test/chromium/oopif.jest.js

```diff
@@ -14,34 +14,47 @@
  * limitations under the License.
  */
 
-const {FFOX, CHROMIUM, WEBKIT, CHANNEL} = require('../utils').testOptions(browserType);
+const {FFOX, CHROMIUM, WEBKIT, CHANNEL} = testOptions;
 
-describe('OOPIF', function() {
-  beforeAll(async function(state) {
-    state.browser = await state.browserType.launch(Object.assign({}, state.defaultBrowserOptions, {
-      args: (state.defaultBrowserOptions.args || []).concat(['--site-per-process']),
-    }));
+registerFixture('sppBrowser', async ({browserType, defaultBrowserOptions}, test) => {
+  const browser = await browserType.launch({
+    ...defaultBrowserOptions,
+    args: (defaultBrowserOptions.args || []).concat(['--site-per-process'])
   });
-  beforeEach(async function(state) {
-    state.context = await state.browser.newContext();
-    state.page = await state.context.newPage();
-  });
-  afterEach(async function(state) {
-    await state.context.close();
-    state.page = null;
-    state.context = null;
-  });
-  afterAll(async function(state) {
-    await state.browser.close();
-    state.browser = null;
-  });
-  it('should report oopif frames', async function({browser, page, server, context}) {
+  try {
+    await test(browser);
+  } finally {
+    await browser.close();
+  }
+});
+
+registerFixture('sppContext', async ({sppBrowser}, test) => {
+  const context = await sppBrowser.newContext();
+  try {
```

Comment:
I see many fixtures do `try {} finally {}`. Shouldn't this be default behavior in the harness?

Reply:
It should, but I often times need to know whether the test succeeded or failed and I don't want to return boolean.